### ISSUE TITLE
S3: Relative path issue

### ISSUE THREAD

**Issue by Luzifer** [ansible/ansible-modules-core#1907]
## Description

Given I have a role doing s3 put operations for files in the `files` directory of that module using relative paths, the command fails with `Local object for PUT does not exist`.
## Ansible Version

``` bash
# cat requirements.txt | grep ansible
ansible==1.8.4
```
## Project structure

(Shortened to exclude some internal processes and to improve overview)

``` bash
# tree
.
├── requirements.txt
├── roles
│   └── vault-init
│       ├── files
│       │   └── core
│       │       └── seal-config
│       └── tasks
│           └── main.yml
├── stacks
│   └── service_vault.yml
└── main.yml
```

In the `main.yml` there is an `include stacks/service_vault.yml` which then includes the role `vault-init`.

The tasks file `vault-init/main.yml` contains the s3 directive:

``` yaml

---
- name: Copy initial files into S3 bucket
  s3:
    bucket: "{{bucket}}"
    mode: put
    object: "{{item}}"
    src: "{{playbook_dir}}/../roles/vault-init/files/{{item}}"
  with_items:
    - core/seal-config
```
## Expected

`src` is `files/{{item}}` or `{{item}}` and the file gets copied to the destination
## Reality

`src` has to contain a workaround like this to access the file as it does not deal with the path relative to the role directory: `{{playbook_dir}}/../roles/vault-init/files/{{item}}`

**Comment by bcoca** [comment]
this is fixed in current devel, just putting src=[{item}} will search in the role's files/ directory for the file first.

**Comment by jvirtanen** [comment]
Should this issue have been fixed in Ansible 2.0.0.2?
